safe/utilities/keyword_io.py

```diff
@@ -117,6 +120,61 @@ def set_keyword_db_path(self, path):
         """
         self.keyword_db_path = str(path)
 
+    @classmethod
+    def read_keywords_file(cls, filename, keyword=None):
```

Comment:
Please add case where the extension is not `.xml` or `.keywords`. Otherwise, LGTM.